core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -164,25 +165,12 @@ public S3RestServiceHandler(@Context ServletContext context)
     mBucketInvalidPrefixPattern = Pattern.compile("^xn--.*");
     mBucketInvalidSuffixPattern = Pattern.compile(".*-s3alias$");
     mBucketValidNamePattern = Pattern.compile("[a-z0-9][a-z0-9\\.-]{1,61}[a-z0-9]");
-
-    // Initiate the S3 API metadata directories
-    if (!mMetaFS.exists(new AlluxioURI(S3RestUtils.MULTIPART_UPLOADS_METADATA_DIR))) {
-      mMetaFS.createDirectory(
-          new AlluxioURI(S3RestUtils.MULTIPART_UPLOADS_METADATA_DIR),
-          CreateDirectoryPOptions.newBuilder()
-              .setRecursive(true)
-              .setMode(PMode.newBuilder()
-                  .setOwnerBits(Bits.ALL)
-                  .setGroupBits(Bits.ALL)
-                  .setOtherBits(Bits.NONE).build())
-              .setWriteType(S3RestUtils.getS3WriteType())
-              .setXattrPropStrat(XAttrPropagationStrategy.LEAF_NODE)
-              .build()
-      );
-    }
-
     mGlobalRateLimiter = (RateLimiter) context.getAttribute(
         ProxyWebServer.GLOBAL_RATE_LIMITER_SERVLET_RESOURCE_KEY);
+    if (((AtomicBoolean) context.getAttribute(
+        ProxyWebServer.MULTIPART_UPLOADS_METADATA_DIR_CREATE_FLAG)).compareAndSet(false, true)) {
```

Comment:
I think only if the create dir request success, we can set it to true.

Reply:
But if mount root as read-only, creating directory will never work. Do we need to determine the status of the root mount?

Reply:
I added the read-only judgment for alluxio root mount.

Reply:
> I added the read-only judgment for alluxio root mount.

Delete it.